jax/scipy/special.py

```diff
@@ -39,6 +39,15 @@ def betaln(x, y):
   return lax.lgamma(x) + lax.lgamma(y) - lax.lgamma(x + y)
 
 
+@_wraps(osp_special.betainc)
+def betainc(a, b, x):
+  a, b, x = _promote_args_inexact("betainc", a, b, x)
+  a = lax._brcast(lax._brcast(a, b), x)  # pylint: disable=protected-access
```

Comment:
I actually want to take this back. You should move this broadcasting back to `lax`. `lax` primitives do usually support broadcasting, just not rank promotion.